packages/grafana-data/src/field/fieldOverrides.ts

```diff
@@ -436,33 +469,90 @@ export const getLinksSupplier =
     fieldScopedVars: ScopedVars,
     replaceVariables: InterpolateFunction,
     timeZone?: TimeZone,
-    dataLinkPostProcessor?: DataLinkPostProcessor
+    dataLinkPostProcessor?: DataLinkPostProcessor,
+    enhancedReplaceVariables?: EnhancedInterpolateFunction
   ) =>
   (config: ValueLinkConfig): Array<LinkModel<Field>> => {
     if (!field.config.links || field.config.links.length === 0) {
       return [];
     }
-
     const linkModels = field.config.links.map((link: DataLink) => {
       const dataContext: DataContextScopedVar = getFieldDataContextClone(frame, field, fieldScopedVars);
+
+      // if we have a row index, expose every field on row as a variable
+      let rowVars: ScopedVars = {};
+      if (config.valueRowIndex !== undefined) {
+        let fieldDisplayValuesProxy: Record<string, DisplayValue> | undefined = getFieldDisplayValuesProxy({
+          frame,
+          rowIndex: config.valueRowIndex,
+        });
+
+        // add all fields in frame as variables
+        frame.fields.forEach((f) => {
+          if (fieldDisplayValuesProxy && fieldDisplayValuesProxy[f.name]) {
+            rowVars[f.name] = {
+              value: fieldDisplayValuesProxy[f.name],
+            };
+          }
+        });
+      }
```

Comment:
Need to figure out how to do this through a variable macro instead (they are lazily evaluated), building every possible var by building a large scopedVar object every time is now how we do it anymore (data macros replaced all of this). https://github.com/grafana/scenes/blob/main/packages/scenes/src/variables/macros/dataMacros.ts